### ISSUE TITLE
loan request shopping cart

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#2006]
ref: https://github.com/ArctosDB/arctos/issues/1875

Should we let users request specific parts for loan?

**Comment by campmlc** [comment]
Yes. And we should consider adding the same information we provide to GGBN
regarding permits for genomic materials so that Arctos can be a searchable
alternative to the GGBN interface.

**Comment by KyndallH** [comment]
Yes! Not just for tissues but skeletal too. How would that work when we enter the skeleton and all they need is the right femur? 

I like having all the permits tied in too. 

Would this just be selecting the part they need or can we incorporate the whole loan request process? Could it be tailored for the collection? Could collections that haven't warmed to the idea, opt out?

**Comment by dustymc** [comment]
>permits

I think that's deserving of a separate Issue - not much technical problem, but exposing permit data (or a lack thereof) is likely a sensitive issue that should be clearly presented to the community without any other clutter.

>tissues but skeletal too

You will never see me trying to make a preemptive distinction between tissues and "vouchers!"

>How would that work when we enter the skeleton and all they need is the right femur?

You tell me, but I suspect this would still involve a fair bit of interaction - eg, they'd enter a comment somewhere, you'd probably want to contact them personally before hucking a bunch of stuff in a box, etc.

>incorporate the whole loan request process?

I think that would be very useful. And this would likely spawn cross-collection requests, so I think it would be based on a project to tie them all together.

>tailored for the collection?

Probably in a limited capacity - I don't think we have the resources to maintain code for every collection, but the documents involved and such would definitely be linked to specific collections.

>opt out

Yea that's probably necessary - no reason to have people fill in forms if the collections aren't willing to accept them.

**Comment by campmlc** [comment]
Regarding permits, we have already presented this to the community in
relation to tissues because of GGBN, and this permit information is going
to be online hopefully very soon. No reason not to include what we already
make available. But perhaps an opt-in button to show permit availability at
the discretion of the collection, so that other types of collections can
choose to add this funcitonality if they want?

**Comment by dustymc** [comment]
>GGBN

That's a very different system from which one can ask very different questions of very different data; I don't think it directly applies to "Arctos Proper."

**Comment by campmlc** [comment]
Because of the UN Convention on Biological Diversity Nagoya Protocol, any
tissues collected after 2012 are required to show permits. That means
Arctos collections are affected, whether or not we use GGBN. We have to be
able to show potential loan recipients that our tissues have permits. I
think it is important that we can do this in Arctos, so I don't have to
tell researchers from the EU to go to the GGBN portal instead.

**Comment by KyndallH** [comment]
The US currently is not part of the Nagoya Protocol, right? So it is only if we deal with one of the 100 something countries that are in it. And how much of the permit do they need? I'm wondering if we should do a refresher for everyone on it because I figure one day, the US will join. Nice to be ahead of the curve when they do join. Beyond the Nagoya Protocol, we should have everything associated with permits and be able to track it.

**Comment by dustymc** [comment]
No US isn't in Nagoya.

If you can find tissues with permits, then some bored intern can find things that should but do not have permits. (Eg all federal lands have required collecting permits since 1906.) I'm happy to open that door, but from the middle of some unrelated Issue that the affected people aren't likely to see.

**Comment by campmlc** [comment]
The EU and Canada are signatories, as are most/many of the developing
countries we have collections from. So if a Canadian or EU researcher wants
a loan of Ecuadorian tissues from MSB, they are required to verify whether
the tissues were collected in according with Nagoya, e.g. with access and
benefits sharing agreements with the source country. So even if the US is
not a signatory, since we house tissues from countries that are, and since
we loan to countries that are, we are affected by it and need to document
that the tissues were collected in accordance with the protocol.

**Comment by dustymc** [comment]
Tabling for apparent lack of interest.

**Comment by campmlc** [comment]
Not lack of interest, just got buried. This would actually be hugely helpful.

**Comment by Jegelewicz** [comment]
Discussed in Object Tracking planning today - any chance we can create a mock-up? How different does this need to be from the internal loan form?

**Comment by campmlc** [comment]
It would need to allow agent data from new agents that don't exist yet in
Arctos, e.g. provide all info to create the agent if the loan is approved.
It would need to provide all the info needed to create a project in Arctos.
It would need to allow specific fields that could be selected by the
collection - for example, a genomics loan request form could be customized
to request tissue type, quantity, and preservation.
Ideally allow for requesting specific catalog records, with alternates
designated.
It would need to allow billing info for the shipment - e.g. a Fedex account
- in a format that would not be saved in Arctos? Is that even possible? I
currently don't fill out all the shipment info in Arctos because I don't
wan't Fedex account numbers saved there.
Notification contacts - who does the loan request go to?
